plasmapy/atomic/ionization_state.py

```diff
@@ -0,0 +1,501 @@
+from typing import Dict, List, Optional, Tuple, Union
```

Comment:
I don't quite understand the split between these two similarly named files. If there really is a lot to do here, maybe a subdirectory within `atomic` would be a better idea?